internal/pkg/cli/env_override.go

```diff
@@ -0,0 +1,152 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package cli
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ssm"
+	"github.com/aws/copilot-cli/internal/pkg/aws/identity"
+	"github.com/aws/copilot-cli/internal/pkg/aws/sessions"
+	"github.com/aws/copilot-cli/internal/pkg/config"
+	"github.com/aws/copilot-cli/internal/pkg/term/prompt"
+	"github.com/aws/copilot-cli/internal/pkg/term/selector"
+	"github.com/aws/copilot-cli/internal/pkg/workspace"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+type overrideEnvOpts struct {
+	*overrideOpts
+
+	// Interfaces to interact with dependencies.
+	ws wsEnvironmentReader
+}
+
+func newOverrideEnvOpts(vars overrideVars) (*overrideEnvOpts, error) {
+	fs := afero.NewOsFs()
+	ws, err := workspace.Use(fs)
+	if err != nil {
+		return nil, err
+	}
+
+	sessProvider := sessions.ImmutableProvider(sessions.UserAgentExtras("env override"))
+	defaultSess, err := sessProvider.Default()
+	if err != nil {
+		return nil, fmt.Errorf("default session: %v", err)
```

Comment:
any reason not `%w`?

Reply:
I remember reading some guidance by @rsc? on how using `%w` increases the surface area of your API because the underlying error percolates up to the client.
So if the error doesn't need to be expected by a client, the guidance was to use `%v` instead. I'm pretty indifferent about it though, I'm happy to change it 🤷